### PR TITLE
Release script upgraded

### PR DESCRIPTION
**Pull request by pastjean**
- Environments are pretty static and we're deploying "versions" that are changing over time but do stays, I do not believe they're intended for what we initialy thought. I've removed them.
- Managing library versions is easier via a git tag so I've added that ability to github action via a `v*` tag in the workflow. The workflow is compatible with with PRs and branches if we ever want to push test versions of dockers and libraries.

## To release

1. Create a tag that has a `vX.X.X` valid semver `git tag -a -m "v0.0.1" v0.0.1`
2. Push the tag `git push --tags`

### CODE REVIEW

**Comment by pankajroark** [comment]
@howespt Would you mind reviewing this PR?

**Review by howespt — COMMENTED** [review]
Have you been able to test these changes?

**Comment by pastjean** [comment]
> Have you been able to test these changes?

Yes i've tested them before disabling the run in prs :)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+name: Release CI
+
+on:
+  push:
+    tags: [ "v*" ]
```
[the remark this answers is not in the document]
Comment (pastjean) [reply .github/workflows/release.yml:5]: We will protect the v* tag to only certain users

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+jobs:
```
[the remark this answers is not in the document]
Comment (pastjean) [reply .github/workflows/release.yml:7]: the secrets are defined https://github.com/basetenlabs/truss/settings/secrets/actions

File: README.md [hunk 3]
```diff
@@ -2,4 +2,15 @@
...
+## Development 
+
+### Release
+
+To release a new version of the library.
+
+1. Create a PR changing the `pyproject.toml` version
+2. Merge the PR
+3. Create a tag `git tag -a -m "vX.X.X" vX.X.X
+4. Push the tag  `git push -u origin vX.X.X`
```
[the remark this answers is not in the document]
Comment (pastjean) [reply README.md:16]: a tag is like a branch, it's a pointer to a commit. So like a branch the parent is a commit

**Review by howespt — APPROVED** [review]

**Pull request merged**